Rosella/src/main/java/me/hydos/rosella/device/VulkanQueue.java

```diff
@@ -0,0 +1,98 @@
+package me.hydos.rosella.device;
+
+import org.lwjgl.vulkan.*;
+
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+public final class VulkanQueue {
+
+    private final VkQueue queue;
+    private final int family;
+
+    private final Lock lock = new ReentrantLock();
+
+    public VulkanQueue(VkQueue queue, int family) {
+        this.queue = queue;
+        this.family = family;
+    }
+
+    public VkQueue getQueue() {
+        return this.queue;
+    }
+
+    public VkDevice getDevice() {
+        return this.queue.getDevice();
+    }
+
+    public int getQueueFamily() {
+        return this.family;
+    }
+
+    public int vkQueueSubmit(VkSubmitInfo submit, long fence) {
+        int result;
+        try {
+            this.lock.lock();
+            result = VK10.vkQueueSubmit(this.queue, submit, fence);
+        } finally {
+            this.lock.unlock();
+        }
+        return result;
+    }
+
+    public int vkQueueSubmit(VkSubmitInfo.Buffer pSubmits, long fence) {
+        int result;
+        try {
+            this.lock.lock();
+            result = VK10.vkQueueSubmit(this.queue, pSubmits, fence);
+        } finally {
+            this.lock.unlock();
+        }
+        return result;
+    }
+
+    public int vkQueueBindSparse(VkBindSparseInfo bindInfo, long fence) {
+        int result;
+        try {
+            this.lock.lock();
+            result = VK10.vkQueueBindSparse(this.queue, bindInfo, fence);
+        } finally {
+            this.lock.unlock();
+        }
+        return result;
+    }
+
+    public int vkQueueBindSparse(VkBindSparseInfo.Buffer pBindInfo, long fence) {
+        int result;
+        try {
+            this.lock.lock();
+            result = VK10.vkQueueBindSparse(this.queue, pBindInfo, fence);
+        } finally {
+            this.lock.unlock();
+        }
+        return result;
+    }
+
+    // THIS IS BAD. VERY VERY BAD. DO NOT DO THIS. EVER... (but if for some reason you do need to here is a safe function)
```

Comment:
use javadoc instead of comments
javadoc examples are in the other classes

Reply:
I used a comment since it wasnt really a documentation of the function

Reply:
use `@Deprecated` on it

Reply:
Done